modules/shared/src/main/scala/org/tessellation/schema/currencyMessage.scala

```diff
@@ -0,0 +1,35 @@
+package org.tessellation.schema
+
+import cats.syntax.semigroup._
+
+import org.tessellation.schema.address.Address
+
+import derevo.cats.{eqv, order, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import enumeratum.values.{StringCirceEnum, StringEnum, StringEnumEntry}
+import eu.timepit.refined.auto.{autoInfer, autoRefineV, autoUnwrap}
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.estatico.newtype.macros.newtype
+
+object currencyMessage {
+  @derive(eqv, show)
+  sealed abstract class MessageType(val value: String) extends StringEnumEntry
+
+  object MessageType extends StringEnum[MessageType] with StringCirceEnum[MessageType] {
+    val values = findValues
+
+    case object Owner extends MessageType("Owner")
+    case object Staking extends MessageType("Staking")
+  }
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class MessageOrdinal(value: NonNegLong) {
+    def next: MessageOrdinal = MessageOrdinal(value |+| 1L)
```

Comment:
instead of `def next`, please provide the `Next` instance (see SnapshotOrdinal for reference). `TransactionOrdinal` looks to be the oldest code and it does not have this refactored yet

Reply:
done